packages/core/src/subgraph/getConfirmDataFromGraph.ts

```diff
@@ -0,0 +1,63 @@
+import { ethers } from "ethers";
+import { GraphQLClient, gql } from 'graphql-request'
+import { config } from "../config.js";
+
+/**
+ * Fetches the confirmation data for multiple challenge assertions from the graph.
+ * 
+ * @param {number[]} assertionIds - An array of assertion IDs for which confirmation data is required.
+ * @returns {Promise<{string[], string}>} A promise that resolves to a tuple containing an array of `bytes32` confirmation data and a `bytes32` confirmation hash.
+ */
+export async function getConfirmDatafromGraph(assertionIds: number[]): Promise<{confirmData: string[], confirmHash: string}> {
+    
+    //initialize graph client
+    const client = new GraphQLClient(config.subgraphEndpoint);
+        
+    //craft query
+    const assertionIdsStrArray: string[] = assertionIds.map(num => num.toString());
+    const assertionIdsStrArrayFormatted = JSON.stringify(assertionIdsStrArray);
+    const query = gql`
+        query NodeConfirmationQuery {
+            nodeConfirmations(where: {id_in: ${assertionIdsStrArrayFormatted}}) {
+                id
+                confirmData
+            }
+        }
+    `
+    console.log(`query: ${query}`);
+
+    //send query
+    const result = await client.request(query) as {
+        nodeConfirmations: [ 
+            {
+                id: string;
+                confirmData: string;
+            }
+        ];
+    };
+
+    //null guard
+    if (!result.nodeConfirmations) {
+        console.log(`Error: nodeConfirmations field is null for subgraph query`);
+        return {confirmData: [], confirmHash: ""};
```

Comment:
This should throw an error if the assertionIds we are passing in don't have events on the subgraph. This error will need to be handled from whoever calls this function.

Remove the console.log, do something like `throw new Error(`NodeConfirmations field is null for subgraph query`)`

Reply:
Removed console log, replaced with throw error